### PR TITLE
Add basic hcloud_subnetwork module

### PR DESCRIPTION
**Pull request by LKaemmerling**
##### SUMMARY

Follow Up from #59366 and Part 2/4 of #59133

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

hcloud_subnetwork
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by ghost — APPROVED** [review]
LGTM

File: lib/ansible/modules/cloud/hcloud/hcloud_subnetwork.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  - hcloud-python >= 1.3.0
+
+extends_documentation_fragment: hcloud
+"""
+
+EXAMPLES = """
+- name: Create a basic subnetwork
+  hcloud_subnetwork:
+    network: my-network
+    ip_range: 10.0.0.0/16
```
Comment (resmo) [inline lib/ansible/modules/cloud/hcloud/hcloud_subnetwork.py:70]: missing required args in examples: type, network_zone

File: lib/ansible/modules/cloud/hcloud/hcloud_subnetwork.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        ip_range:
+            description: IP range of the Network
+            type: str
+            returned: always
+            sample: 10.0.0.0/8
+        type:
+            description: Type of subnetwork
+            type: str
+            returned: always
+            sample: eu-central
```
Comment (resmo) [inline lib/ansible/modules/cloud/hcloud/hcloud_subnetwork.py]: copy paste error from zone?

File: lib/ansible/modules/cloud/hcloud/hcloud_subnetwork.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+DOCUMENTATION = """
+---
+module: hcloud_subnetwork
+
+short_description: Create and manage cloud subnetworks on the Hetzner Cloud.
+
+version_added: "2.9"
+
+description:
+    - Create, update and manage cloud subnetworks on the Hetzner Cloud.
```
Comment (resmo) [inline lib/ansible/modules/cloud/hcloud/hcloud_subnetwork.py]: Suggestion: Create, update and __delete__ cloud subnetworks on the Hetzner Cloud.

File: lib/ansible/modules/cloud/hcloud/hcloud_subnetwork.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = """
+---
+module: hcloud_subnetwork
+
+short_description: Create and manage cloud subnetworks on the Hetzner Cloud.
```
Comment (resmo) [inline lib/ansible/modules/cloud/hcloud/hcloud_subnetwork.py]: Suggestion: Manage cloud subnetworks on the Hetzner Cloud.

File: lib/ansible/modules/cloud/hcloud/hcloud_subnetwork.py [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+version_added: "2.9"
+
+description:
+    - Create, update and manage cloud subnetworks on the Hetzner Cloud.
+    - You need at least hcloud-python 1.3.0.
```
Comment (resmo) [inline lib/ansible/modules/cloud/hcloud/hcloud_subnetwork.py]: This information is redundant, I would remove it

**Review by resmo — COMMENTED** [review]
good job, some minors in docs found

**Review by resmo — APPROVED** [review]
shipit

**Pull request merged**